src/rgw/rgw_op.h

```diff
@@ -422,34 +425,7 @@ class RGWGetObj : public RGWOp {
 
   int init_common();
 public:
-  RGWGetObj() {
-    range_str = NULL;
-    if_mod = NULL;
-    if_unmod = NULL;
-    if_match = NULL;
-    if_nomatch = NULL;
-    mod_zone_id = 0;
-    mod_pg_ver = 0;
-    start = 0;
-    ofs = 0;
-    total_len = 0;
-    end = -1;
-    mod_ptr = NULL;
-    unmod_ptr = NULL;
-    get_data = false;
-    partial_content = false;
-    range_parsed = false;
-    skip_manifest = false;
-    is_slo = false;
-    first_block = 0;
-    last_block = 0;
-    q_ofs = 0;
-    q_len = 0;
-    first_data = true;
-    cur_ofs = 0;
-    get_retention = false;
-    get_legal_hold = false;
- }
+  RGWGetObj() = default;
```

Comment:
in general, i do prefer in-class member initializers to constructors

> Moving all default values to member declarations

did you forget to do this part? this leaves many of the integer/boolean/pointer members uninitialized